src/scripts/main.js

```diff
@@ -1,3 +1,243 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const field = document.querySelector('tbody');
+const scoreField = document.querySelector('.game-score');
+
+let state = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+let newState;
+let isWin = false;
+let score = 0;
+const fieldMatrix = [...field.children].map(row => [...row.children]);
+
+button.addEventListener('click', buttonHandler);
+document.addEventListener('keydown', move);
+
+function buttonHandler() {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.innerText = 'Restart';
+    messageStart.classList.add('hidden');
+  } else {
+    state = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    score = 0;
+    messageLose.classList.toggle('hidden', true);
+    messageWin.classList.toggle('hidden', true);
+  }
+
+  addTile();
+  addTile();
+  render();
+}
+
+function addTile() {
+  newState = state.slice();
+
+  let changed = false;
+
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      if (state[i][j] !== newState[i][j]) {
+        changed = true;
+        break;
+      }
+    }
+
+    if (changed) {
+      break;
+    }
+  }
+
+  if (!changed) {
+    const [y, x] = findEmptyCell();
+
+    state[y][x] = randomNumber();
+  }
+}
+
+function findEmptyCell() {
+  const emptyCells = [];
+
+  state.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      if (cell === 0) {
+        emptyCells.push([rowIndex, columnIndex]);
+      }
+    });
+  });
+
+  return emptyCells[Math.floor(Math.random() * emptyCells.length)];
+}
+
+function randomNumber() {
+  return Math.random() >= 0.9 ? 4 : 2;
+}
+
+function render() {
+  state.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      const elem = fieldMatrix[rowIndex][columnIndex];
+
+      elem.classList = 'field-cell';
+
+      if (cell === 0) {
+        elem.innerText = '';
+      } else {
+        elem.innerText = cell;
+        elem.classList.add(`field-cell--${cell}`);
+      }
+    });
+  });
+
+  scoreField.innerText = score;
+}
+
+function move(e) {
+  newState = state;
+
+  switch (e.key) {
+    case 'ArrowLeft':
```

Comment:
move ArrowLeft, ArrowRight, ArrowDown and ArrowUp to the constants as it will make a code cleaner